pkg/controller/certificates/rootcacertpublisher/publisher.go

```diff
@@ -228,6 +303,55 @@ func (c *Publisher) syncNamespace(ctx context.Context, ns string) (err error) {
 	return err
 }
 
+func (c *Publisher) syncClusterTrustBundle(ctx context.Context, _ string) (err error) {
+	startTime := time.Now()
+	defer func() {
+		recordMetrics(startTime, clusterTrustBundleResource, err)
+		klog.FromContext(ctx).V(4).Info("Finished syncing ClusterTrustBundle", "signerName", c.signerName, "elapsedTime", time.Since(startTime))
+	}()
+
+	signerTrustBundles, err := c.ctbLister.List(labels.Everything())
+	switch {
+	case apierrors.IsNotFound(err) || (err == nil && len(signerTrustBundles) == 0):
+		_, err := c.client.CertificatesV1alpha1().ClusterTrustBundles().Create(ctx, &certificatesv1alpha1.ClusterTrustBundle{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "kubernetes.io:kube-apiserver-serving:serving-trust",
```

Comment:
Are we guaranteed to have leader election such that this will never have more than one loop running?  It would be silly to have different instance fighting to converge to a single item.

Would it be simpler to make the name deterministic based on the root CA contents?

Reply:
Currently the `Publisher`'s only used in KCM which I think always runs behind a leader election. However, if it were used outside that context, we may not always have that :neutral_face:

I suppose we could just hash the bundle and append the result to a common prefix in a similar fashion to how GenerateName is currently used.